### ISSUE TITLE
Help with Install ...

### ISSUE THREAD

**Issue by LeeUnd** [danpros/htmly#588]
When I run the "install.php" file, iI get the error code, "no permission to write in the Directory." I am not sure what directory it's speaking of but the entire directory for this website is owned by me. File permissions are 777, 775, & 664. Do I need to change something?

I have another website with HTMLy on the same disk/system and it works fine! Suggestions?

**Comment by JPLeRouzic** [comment]
Weird, there is no "_no permission to write in the Directory_." message in HTMLy, so it comes from some other tool.

**Comment by LeeUnd** [comment]
Wow! Okay, I'll try something else. Thanks for the quick response.

**Comment by LeeUnd** [comment]
Any idea how to find out what is causing this?

**Comment by LeeUnd** [comment]
I just noticed it say it in the "install.php" file.

**Comment by JPLeRouzic** [comment]
I am sorry there is this message in install.php.
I made a quick check with the search function of Github and it returned nothing.
I apologize.

**Comment by LeeUnd** [comment]
That's okay. I just can't figure this out. It worked fine before. ...

**Comment by bttrx** [comment]
Did you also check **directory** permissions for your HTMLy root directory?

**Comment by LeeUnd** [comment]
Yes. I have checked the permissions for all the directories, and files.

**Comment by JPLeRouzic** [comment]
This message "no permission to write in the Directory.'" is in the function testTheEnvironment() in install.php
It is sent when the root of your site is not writable by Apache/PHP.

It has nothing to do with **your** own rights but the **rights of Apache**.

In this picture from my web site you can see that Apache has access to my ./ folder (www-data).

![image](https://user-images.githubusercontent.com/18621529/201035759-759a3e52-7768-4ae2-a9f5-3f8e0d69174b.png)

**Comment by LeeUnd** [comment]
Okay. Thanks. Now how do I fix it?

**Comment by JPLeRouzic** [comment]
> Okay. Thanks. Now how do I fix it?

I do not know characteristics of your host, there are hundred of them with may different configurations.

For myself I have a VPS at OVH with Debian pre-installed. I installed the HTTP webserver so I know how it works.

You might have something very different, for example a [web hosting control panel](https://en.wikipedia.org/wiki/Web_hosting_control_panel) like cpanel, plesk, etc. **In this case I can't help you.**

**I can't help you** either if you use another HTTP server like Ningx.

 **I can't help you** either if you use another OS like Windows.

On a Linux system, like Debian, the HTTP server is a user in itself. So it has access rights to the file system as any other users.
 This is tricky, for example if I try to list "my" file permissions on the directory of my site, I have no right to access them. This is the good secure procedure, _in contradiction to what is written on most tutorials on Internet_. Only the HTTP server can access this folder. BTW PHP is just a module (a program) that is run by HTTP server, so PHP is not a user in Linux world.

Usually Apache is installed with the identifier "www-data" but the person who installed Apache could have chosen something else.

As a super user you assigned rights to www-data to access this folder.
Assign ownership of your website directory to www-data:
`chown -R www-data:www-data /var/www/html/domain.com`

Then set proper permission to the website directory.
`chmod -R 755 /var/www/html/domain.com`

You can follow this tutorial (I didn't check it carefully)
https://cloudinfrastructureservices.co.uk/how-to-install-apache-web-server-on-debian-11/

**Comment by LeeUnd** [comment]
I wasn't able to get it to work locally, but it installed just fine on my web server.,
